src/clincoded/static/components/collection.js

```diff
@@ -36,370 +49,547 @@ var Collection = module.exports.Collection = createReactClass({
-    var tds = row.cells.map( function (cell, index) {
-        var cellValue = (typeof cell.value === 'object') ? cell.value.join(', ') : cell.value;
-        if (index === 0) {
-            return (
-                <td key={index}><a href={row.item['@id']}>{cellValue}</a></td>
-            );
-        }
-        return (
-            <td key={index}>{cellValue}</td>
-        );
-    });
-    return (
-        <tr key={id} hidden={props.hidden} data-href={id}>{tds}</tr>
-    );
-};
-
-var Table = module.exports.Table = createReactClass({
+const Table = module.exports.Table = createReactClass({
+    mixins: [FormMixin, RestMixin],
+    
     contextTypes: {
         fetch: PropTypes.func
     },
 
+    propTypes: {
+        href: PropTypes.string,
+        columns: PropTypes.array,
+        href_url: PropTypes.object,
+        context: PropTypes.object,
+        defaultSortOn: PropTypes.number
+    },
+
+
     getDefaultProps: function () {
         return {
             defaultSortOn: 0
         };
     },
 
     getInitialState: function () {
-        var state = this.extractParams(this.props);
-        state.columns = this.guessColumns(this.props);
-        state.data = new Data([]);  // Tables may be long so render empty first
-        state.communicating = true;
-        return state;
+        const { sortColumn, reversed, searchTerm } = this.extractParams(this.props);
+        return {
+            sortColumn,
+            reversed,
+            searchTerm,
+            rows: [],
+            columns: [],
+            communicating: true,
+            modifiedUsers: [],
+            allRequest: '',
+            submitBusy: false,
+            updateMsg: null,
+        };
+    },
+
+    componentDidMount: function () {
+        const columns = this.getColumns(this.props);
+        this.fetchAll(this.props, columns);
     },
 
     componentWillReceiveProps: function (nextProps) {
-        var updateData = false;
+        let updateData = false;
         if (nextProps.context !== this.props.context) {
             updateData = true;
-            this.setState({
-                communicating: this.fetchAll(nextProps)
-            });
+            this.fetchAll(nextProps);
         }
         if (nextProps.columns !== this.props.columns) {
             updateData = true;
         }
         if (updateData) {
-            var columns = this.guessColumns(nextProps);
+            const columns = this.getColumns(nextProps);
             this.extractData(nextProps, columns);
         }
         if (nextProps.href !== this.props.href) {
-            this.extractParams(nextProps);
+            const { sortColumn, reversed, searchTerm } = this.extractParams(nextProps);
+            this.setState({ sortColumn, reversed, searchTerm });
         }
+    },
 
+    componentWillUnmount: function () {
+        if (typeof this.submitTimer != 'undefined') {
+            clearTimeout(this.submitTimer);
+        }
+        var request = this.state.allRequest;
```

Comment:
Does `let` not work here? I changed it locally and didn't run into any issues, let me know otherwise!

Reply:
It probably works. I just moved this function from the bottom of the file up to here. I didn't change anything other than that. It should probably be `const` actually.